File: app/src/main/java/com/sachtech/datingapp/ui/home/adapter/ProfileItemListAdapter.java
// ProfileItemListAdapter.java
package com.sachtech.datingapp.ui.home.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.sachtech.datingapp.R;
import com.sachtech.datingapp.R.id;
import com.sachtech.datingapp.ui.profile.listener.OnItemSelected;
import com.sachtech.datingapp.utils.PrefKeys;
import com.sachtech.datingapp.utils.Preferences;
import com.sachtech.datingapp.utils.ProfileItem;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

import kotlin.jvm.internal.Intrinsics;


public class ProfileItemListAdapter extends RecyclerView.Adapter<ProfileItemListAdapter.ViewHolder> {
    private int selected_position;
    @Nullable
    private ArrayList itemList;
    @NotNull
    private final Context context;
    @NotNull
    private final ArrayList stringlist;
    @Nullable
    private final String item;
    @Nullable
    private final String selectedType;
    @NotNull
    private final com.sachtech.datingapp.ui.profile.listener.OnItemSelected onItemSelected;

    @Nullable
    public final ArrayList getItemList() {
        return this.itemList;
    }

    public final void setItemList(@Nullable ArrayList var1) {
        this.itemList = var1;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_profile_items, parent, false));
    }

   public void onBindViewHolder(@NotNull final ViewHolder holder, final int position) {
        this.itemList = new ArrayList();
        View var10000 = holder.itemView;
        TextView var3 = (TextView) var10000.findViewById(id.item_profile);
        var3.setText((CharSequence) this.stringlist.get(position));
        if (Intrinsics.areEqual(this.selectedType, "single")) {
            if (holder.getAdapterPosition() == this.selected_position) {
                var10000 = holder.itemView;
                ((TextView) var10000.findViewById(id.item_profile)).setBackgroundResource(R.drawable.dialog_background);
                var10000 = holder.itemView;
                ((TextView) var10000.findViewById(id.item_profile)).setTextColor(ContextCompat.getColor(this.context, R.color.colorWhite
                ));
            } else {
                var10000 = holder.itemView;
                ((TextView) var10000.findViewById(id.item_profile)).setBackgroundResource(R.drawable.dialog_unselected);
                var10000 = holder.itemView;
                ((TextView) var10000.findViewById(id.item_profile)).setTextColor(ContextCompat.getColor(this.context, R.color.colorGrey));
            }

            holder.itemView.setOnClickListener((OnClickListener) (new OnClickListener() {
                public final void onClick(View it) {
                    String var2 = ProfileItemListAdapter.this.getItem();
                    Preferences var10000;
                    String var10001;
                    if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getORIGIN())) {
                        var10000 = Preferences.INSTANCE;
                        var10001 = PrefKeys.INSTANCE.getORIGIN();
                        var10000.setPref(var10001, ProfileItemListAdapter.this.getStringlist().get(position));
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getCOLORCOMPLEXION())) {
                        var10000 = Preferences.INSTANCE;
                        var10001 = PrefKeys.INSTANCE.getCOLORCOMPLEXION();
                        var10000.setPref(var10001, ProfileItemListAdapter.this.getStringlist().get(position));
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getHAIRCOLOR())) {
                        var10000 = Preferences.INSTANCE;
                        var10001 = PrefKeys.INSTANCE.getHAIRCOLOR();
                        var10000.setPref(var10001, ProfileItemListAdapter.this.getStringlist().get(position));
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getEYECOLOR())) {
                        var10000 = Preferences.INSTANCE;
                        var10001 = PrefKeys.INSTANCE.getEYECOLOR();
                        var10000.setPref(var10001, ProfileItemListAdapter.this.getStringlist().get(position));
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getPROFESSION())) {
                        var10000 = Preferences.INSTANCE;
                        var10001 = PrefKeys.INSTANCE.getPROFESSION();
                        var10000.setPref(var10001, ProfileItemListAdapter.this.getStringlist().get(position));
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getEDUCATION())) {
                        Preferences.INSTANCE.setPref("education", ProfileItemListAdapter.this.getStringlist().get(position));
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getNATIONALITY())) {
                        Preferences.INSTANCE.setPref("nationality", ProfileItemListAdapter.this.getStringlist().get(position));
                    }

                    com.sachtech.datingapp.ui.profile.listener.OnItemSelected var3 = ProfileItemListAdapter.this.getOnItemSelected();
                    Object var4 = ProfileItemListAdapter.this.getStringlist().get(position);
                    var3.selectedItem((String) var4);
                    ProfileItemListAdapter.this.selected_position = holder.getAdapterPosition();
                    ProfileItemListAdapter.this.notifyDataSetChanged();
                }
            }));
        } else if (Intrinsics.areEqual(this.selectedType, "multiple")) {
            var10000 = holder.itemView;
           ((TextView) var10000.findViewById(id.item_profile)).setBackgroundResource(R.drawable.dialog_unselected);
            var10000 = holder.itemView;
            ((TextView) var10000.findViewById(id.item_profile)).setTextColor(ContextCompat.getColor(this.context, R.color.colorGrey));
            holder.itemView.setOnClickListener((OnClickListener) (new OnClickListener() {
                public final void onClick(View it) {
                    View var10000 = holder.itemView;
                    ((TextView) var10000.findViewById(id.item_profile)).setBackgroundResource(R.drawable.dialog_background);
                    var10000 = holder.itemView;
                    ((TextView) var10000.findViewById(id.item_profile)).setTextColor(ContextCompat.getColor(ProfileItemListAdapter.this.getContext(), R.color.colorWhite));
                    ArrayList var3 = ProfileItemListAdapter.this.getItemList();
                    if (var3 == null) {
                        Intrinsics.throwNpe();
                    }

                    View var10001 = holder.itemView;
                    TextView var5 = (TextView) var10001.findViewById(id.item_profile);
                    var3.add(var5.getText().toString());
                    String var2 = ProfileItemListAdapter.this.getItem();
                    ArrayList var10002;
                    Preferences var4;
                    String var6;
                    if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getCOLORCOMPLEXION())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_COLORCOMPLEXION();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getHAIRCOLOR())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_HAIRCOLOR();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getEYECOLOR())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_EYECOLOR();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getPROFESSION())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_PROFESSION();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getEDUCATION())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_EDUCATION();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getQURAN())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_QURAN();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getPRAYS())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_PRAYS();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getHALAL())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_HALAL();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getWEAR())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_WEAR();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getLIFEAFTERMARRIAGE())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_LIFE_AFTER_MARRIAGE();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ? joinToString(var10002): null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getMARRIAGEPLANS())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_MARRIAGE_PLANS();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ?joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getREVERT())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_REVERT();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                                var4.setPref(var6, var10002 != null ?joinToString(var10002) : null);
                    } else if (Intrinsics.areEqual(var2, ProfileItem.INSTANCE.getBEARD())) {
                        var4 = Preferences.INSTANCE;
                        var6 = PrefKeys.INSTANCE.getPREF_BEARD();
                        var10002 = ProfileItemListAdapter.this.getItemList();
                        var4.setPref(var6, var10002 != null ?joinToString(var10002) : null);
                    }

                    OnItemSelected var7 = ProfileItemListAdapter.this.getOnItemSelected();
                    ArrayList var8 = ProfileItemListAdapter.this.getItemList();
                    var6 = var8 != null ? joinToString(var8):null;
                    var7.selectedItem(var6);
                }
            }));
        }

    }

    public String joinToString(ArrayList arrayList) {
        StringBuilder builder = new StringBuilder("");
        for (int i = 0; i < arrayList.size(); i++) {
            builder.append(arrayList).append(",");
        }
        String strList = builder.toString();
        if (strList.length() > 0)
            strList = strList.substring(0, strList.length() - 1);
        return strList;
    }

    public int getItemCount() {
        return this.stringlist.size();
    }

    @NotNull
    public final Context getContext() {
        return this.context;
    }

    @NotNull
    public final ArrayList getStringlist() {
        return this.stringlist;
    }

    @Nullable
    public final String getItem() {
        return this.item;
    }

    @Nullable
    public final String getSelectedType() {
        return this.selectedType;
    }

    @NotNull
    public final com.sachtech.datingapp.ui.profile.listener.OnItemSelected getOnItemSelected() {
        return this.onItemSelected;
    }

    public ProfileItemListAdapter(@NotNull Context context, @NotNull ArrayList stringlist, @Nullable String item, @Nullable String selectedType, @NotNull com.sachtech.datingapp.ui.profile.listener.OnItemSelected onItemSelected) {
        super();
        this.context = context;
        this.stringlist = stringlist;
        this.item = item;
        this.selectedType = selectedType;
        this.onItemSelected = onItemSelected;
        this.selected_position = -1;
    }


    public class ViewHolder extends RecyclerView.ViewHolder {
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/sachtech/datingapp/ui/home/adapter/ProfileItemListAdapter.kt
/*
package com.sachtech.datingapp.ui.home.adapter

import android.content.Context
import androidx.core.content.ContextCompat

import com.sachtech.datingapp.R
import com.sachtech.datingapp.utils.PrefKeys
import com.sachtech.datingapp.utils.Preferences.setPref
import com.sachtech.datingapp.utils.ProfileItem
import com.sachtech.datingapp.utils.setPref
import gurtek.mrgurtekbase.adapter.BaseAdapter
import kotlinx.android.synthetic.main.item_profile_items.view.*

class ProfileItemListAdapter(
    val context: Context,
    val stringlist: ArrayList<String>,
    val item: String?,
    val selectedType: String?,
    val onItemSelected: OnItemSelected
) :
    BaseAdapter<String>(R.layout.item_profile_items) {
    private var selected_position = -1

    var itemList:ArrayList<String>?=null
    override fun onBindViewHolder(holder: IViewHolder, position: Int) {
        itemList= ArrayList()
        holder.itemView.item_profile.text = stringlist[position]
        if(selectedType=="single") {
            if (holder.adapterPosition == selected_position) {
                holder.itemView.item_profile.setBackgroundResource(R.drawable.dialog_background)
                holder.itemView.item_profile.setTextColor(
                    ContextCompat.getColor(
                        context,
                        R.color.colorWhite
                    )
                )
            } else {
                holder.itemView.item_profile.setBackgroundResource(R.drawable.dialog_unselected)
                holder.itemView.item_profile.setTextColor(
                    ContextCompat.getColor(
                        context,
                        R.color.colorGrey
                    )
                )
            }
            holder.itemView.setOnClickListener {
                when (item) {
                    ProfileItem.ORIGIN -> setPref(PrefKeys.ORIGIN, stringlist[position])
                    ProfileItem.COLORCOMPLEXION -> setPref(
                        PrefKeys.COLORCOMPLEXION,
                        stringlist[position]
                    )
                    ProfileItem.HAIRCOLOR -> setPref(PrefKeys.HAIRCOLOR, stringlist[position])
                    ProfileItem.EYECOLOR -> setPref(PrefKeys.EYECOLOR, stringlist[position])
                    ProfileItem.PROFESSION -> setPref(PrefKeys.PROFESSION, stringlist[position])
                    ProfileItem.EDUCATION -> setPref(PrefKeys.EDUCATION, stringlist[position])
                    ProfileItem.NATIONALITY -> setPref(PrefKeys.NATIONALITY, stringlist[position])
                }
                onItemSelected.selectedItem(stringlist[position])
                selected_position = holder.adapterPosition
                notifyDataSetChanged()
            }
        }
        else if(selectedType=="multiple") {
            holder.itemView.item_profile.setBackgroundResource(R.drawable.dialog_unselected)
            holder.itemView.item_profile.setTextColor(
                ContextCompat.getColor(
                    context,
                    R.color.colorGrey
                )
            )

            holder.itemView.setOnClickListener {
                holder.itemView.item_profile.setBackgroundResource(R.drawable.dialog_background)
                holder.itemView.item_profile.setTextColor(
                    ContextCompat.getColor(
                        context,
                        R.color.colorWhite
                    )
                )
                itemList!!.add(holder.itemView.item_profile.text.toString())
                when (item) {
                    ProfileItem.COLORCOMPLEXION -> setPref(
                        PrefKeys.PREF_COLORCOMPLEXION,
                        itemList?.joinToString(",")
                    )
                    ProfileItem.HAIRCOLOR -> setPref(
                        PrefKeys.PREF_HAIRCOLOR,
                        itemList?.joinToString(",")
                    )
                    ProfileItem.EYECOLOR -> setPref(PrefKeys.PREF_EYECOLOR, itemList?.joinToString(","))
                    ProfileItem.PROFESSION -> setPref(PrefKeys.PREF_PROFESSION, itemList?.joinToString(","))
                    ProfileItem.EDUCATION -> setPref(PrefKeys.PREF_EDUCATION, itemList?.joinToString(","))
                    ProfileItem.QURAN -> setPref(PrefKeys.PREF_QURAN, itemList?.joinToString(","))
                    ProfileItem.PRAYS -> setPref(PrefKeys.PREF_PRAYS, itemList?.joinToString(","))
                    ProfileItem.HALAL -> setPref(PrefKeys.PREF_HALAL, itemList?.joinToString(","))
                    ProfileItem.WEAR -> setPref(PrefKeys.PREF_WEAR, itemList?.joinToString(","))
                    ProfileItem.LIFEAFTERMARRIAGE -> setPref(PrefKeys.PREF_LIFE_AFTER_MARRIAGE, itemList?.joinToString(","))
                    ProfileItem.MARRIAGEPLANS -> setPref(PrefKeys.PREF_MARRIAGE_PLANS, itemList?.joinToString(","))
                    ProfileItem.REVERT -> setPref(PrefKeys.PREF_REVERT, itemList?.joinToString(","))
                    ProfileItem.BEARD -> setPref(PrefKeys.PREF_BEARD, itemList?.joinToString(","))
                }
                onItemSelected.selectedItem(itemList?.joinToString(",")!!)
            }
        }

    }

    override fun getItemCount(): Int {
        return stringlist.size
    }

}

interface OnItemSelected{
    fun selectedItem(item:String)
}


*/
